src/mainwindow.cpp

```diff
@@ -4439,12 +4439,12 @@ void MainWindow::on_modList_customContextMenuRequested(const QPoint &pos)
     m_ContextRow = m_ContextIdx.row();
 
     QMenu *menu = nullptr;
-    QMenu *allMods = modListContextMenu();
     if (m_ContextRow == -1) {
       // no selection
-      menu = allMods;
+      menu = modListContextMenu(this);

```

Comment:
This is still a child of something. Are we sure that we can delete it at the end of the function without it being automatically deleted again later on when the main window is closed? It's probably better to make the parent null.

Reply:
It should be fine. The Qt deconstructor will detach it. But again, I'm not sure why these need to use pointers in the first place unless we are storing them long-term.